[TACHYON-833] Implement Journal crash test out of unit test

swatDong opened the pull request:
This PR implements a util to test Journal when crashing. The test can be run by `bin/tachyon jCrashTest [args]`. Following is the workflow:

    i. Start Master and Clients to request to Master.
    ii. Kill Master.
    iii. Restart Master and check status.

This test is at client side and is put into the examples package. I'm going to add more operations and configurable settings.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4623/<h2>Build result: FAILURE</span></h2>[...truncated 2946 lines...][JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/pom.xml to org.tachyonproject/tachyon-clients/0.8.0-SNAPSHOT/tachyon-clients-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/pom.xml to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/pom.xml to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/pom.xml to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/pom.xml to org.tachyonproject/tachyon-parent/0.8.0-SNAPSHOT/tachyon-parent-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/dependency-reduced-pom.xml to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/pom.xml to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jarchannel stopped
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4624/<h2>Build result: FAILURE</span></h2>[...truncated 2946 lines...][JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/pom.xml to org.tachyonproject/tachyon-clients/0.8.0-SNAPSHOT/tachyon-clients-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/pom.xml to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/pom.xml to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/pom.xml to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/pom.xml to org.tachyonproject/tachyon-parent/0.8.0-SNAPSHOT/tachyon-parent-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/dependency-reduced-pom.xml to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/pom.xml to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jarchannel stopped
Test FAILed.

gpang commented:
Jenkins, test this please.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4643/
Test FAILed.

apc999 commented:
Jenkins, test this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4663/
Test PASSed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4828/<h2>Build result: FAILURE</span></h2>[...truncated 3297 lines...][JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/pom.xml to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/pom.xml to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/pom.xml to org.tachyonproject/tachyon-parent/0.8.0-SNAPSHOT/tachyon-parent-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/dependency-reduced-pom.xml to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/pom.xml to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jarchannel stopped
Test FAILed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4829/
Test PASSed.

gpang commented on bin/tachyon:
```diff
@@ -16,6 +16,7 @@ function printUsage {
   echo -e "  loadufs       \t Load existing files in underlayer filesystem into Tachyon."
   echo -e "  runTest       \t Run a end-to-end test on a Tachyon cluster."
   echo -e "  runTests      \t Run all end-to-end tests on a Tachyon cluster."
+  echo -e "  jCrashTest    \t Test the Master Journal System in a crash scenario"
```
This should actually say "journal" instead of just "j". Also, does it make sense to have this new command, or would it fit under runTest?

Nevermind, it looks like it stops existing server and creates a new one. That info should be in the "help" of the journal test.

swatDong replied:
I think it should be separated with runTest because this test needs to start/stop Master.
I'll change it to 'journalCrashTest' and update the 'help'.

gpang commented on examples/pom.xml:
```diff
@@ -19,6 +19,11 @@
 
   <dependencies>
     <dependency>
+      <groupId>commons-cli</groupId>
```
Does tachyon already depend on commons-cli elsewhere?

swatDong replied:
No. I haven't found commons-cli in Tachyon's pom or jar files.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+import tachyon.util.CommonUtils;
+
+/**
+ * Class to perform Journal crash test. Launch clients to generate the Journal and check status.
```
the clients issue commands to the master, and the master generates journal events.

swatDong replied:
Thanks. I'll update the comments.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+    private final ClientOpType mOpType;
+    /** The Tachyon Client hold by this thread. */
+    // TODO: use TachyonFileSystem instead of the deprecated TachyonFS
+    private final TachyonFS mTfs;
```
Can you use the new client?

swatDong replied:
Yes. I'll fix the current issues, address the comments, then use the new client.
After that I'll remove the 'WIP' tag of this PR.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+          //    break;
+          //  }
+          //}
+          mSuccessNum++;
```
please add a space before the "++".

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+    }
+  }
+
+
```
extra blank line?

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+
+  // The two Exit Codes are used to tell script if the test runs well.
+  private static final int EXIT_FAILED = 99;
+  private static final int EXIT_SUCCESS = 100;
```
why are these status codes this way? Why not -1 and 0 or something more standard?

swatDong replied:
The 'bin/tachyon' script uses `ret > 128` to determine if the process is terminated due to signal.
If the status code is -1 (`System.exit(-1)`), `ret` in script will be 255. The status codes here should be not greater than 128.

To be more standard, I'll use 0 for success and 1 for failed.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+  private static TachyonURI sMasterAddress = null;
+  private static String sTestDir = null;
+  private static List<Thread> sClientThreadList = null;
+  private static List<ClientThread> sClientThreads = null;
```
what is the difference between sClientThreads and sClientThreadList?

swatDong replied:
One is used to wait all the threads finished (`thread.join()`). Another is used to get the result (`clientThread.getSuccessNum()`).

I can use `ClientThread extends Thread` to only have one list.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+    if (args.length < 4) {
+      System.out.println("java -cp tachyon-" + Version.VERSION + "-jar-with-dependencies.jar "
+          + "tachyon.examples.JournalCrashTest " + "<0:Run Clients|1: Check Status> "
+          + "<TachyonMasterAddress> <TestTachyonDir> <TmpOutFile> [-options]");
```
why are some options in the command line, and some in "-options"?

swatDong replied:
My previous idea is to put the optional args in "-options".
I'll put all the args in "-options" with some default values.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,377 @@
+      System.exit(EXIT_FAILED);
+    }
+
+    int runType = Integer.parseInt(args[0]);
```
Why are there two different modes? Why doesn't the test run the clients, and then check the status at the end?

swatDong replied:
Between the two modes there is a step to kill the Master, which should be done by script.
If needed, I can make the test be aware of Master crash and run the test in one mode.

gpang commented:
Thanks for this test! I left some comments in the code.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4861/
Test PASSed.

swatDong commented:
Two more steps:
i. make all the args in "-options"
ii. use the new tachyon.client.file.TachyonFileSystem

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/4967/
Test PASSed.

swatDong commented:
@gpang I've addressed the comments. And make this test
  1) run clients and check status in one program
  2) put all options in [-options]

gpang commented on bin/tachyon:
```diff
@@ -116,6 +117,59 @@ function runTest {
 
+function journalCrashTest {
+  if [ "$#" -lt 1 ]; then
+    echo "Usage: tachyon journalCrashTest <test-duration-in-seconds> [-options]"
```
Is there a way to leverage the commons-cli library to print out the usage automatically? Having it print out in both this file and the .java file seems error prone to keep consistent.

Also, why is the test duration not in the cli options?

swatDong replied:
I'll use HelpFormatter.printHelp to print out the usage and only put it in the java file.
Then, I'll kill/restart Master for multiple times in the java file instead of a specified time duration.

gpang commented on bin/tachyon:
```diff
@@ -116,6 +117,59 @@ function runTest {
+  $bin/tachyon-stop.sh
+  # If test process received a signal abort early
+  # Exit Codes for signals terminating a JVM are 128 + signal
+  if [ $RET -gt 128 ]; then
```
Does this line still work when the status changed to 0 and 1?

swatDong replied:
Yes. If the test process is aborted by a signal, this line will work.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,311 @@
+ * Class to perform Journal crash test. The clients issue commands to the master, and the master
+ * generates journal events. Check if the master can generate and reproduce the journal correctly.
+ */
+public class JournalCrashTest {
```
Probably not in this PR, but it would be great to have this test kill/restart the master multiple times during the duration of the test.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,311 @@
+
+    // Restart Tachyon Master.
+    String restartMasterCommand = new TachyonConf().get(Constants.TACHYON_HOME)
+        + "/bin/tachyon-start.sh master";
```
It is a little confusing that the master is started in the bash script, but also started here. Why is the master started in both places?

Ideally, it should only be started/killed in one place.

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,311 @@
+  }
+
+  private static void printUsage() {
+    System.out.println("Usage: java -cp tachyon-" + Version.VERSION + "-jar-with-dependencies.jar "
```
Is there an automatic way for the options to print out the usage, instead of having to do it manually?

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5001/<h2>Build result: FAILURE</span></h2>[...truncated 3343 lines...][JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/pom.xml to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/pom.xml to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/pom.xml to org.tachyonproject/tachyon-parent/0.8.0-SNAPSHOT/tachyon-parent-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/dependency-reduced-pom.xml to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/pom.xml to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jarchannel stopped
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5005/<h2>Build result: FAILURE</span></h2>[...truncated 3427 lines...][JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/unshaded/target/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client-unshaded/0.8.0-SNAPSHOT/tachyon-client-unshaded-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/pom.xml to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/servers/target/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-servers/0.8.0-SNAPSHOT/tachyon-servers-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/pom.xml to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/hdfs/target/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-hdfs/0.8.0-SNAPSHOT/tachyon-underfs-hdfs-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/pom.xml to org.tachyonproject/tachyon-parent/0.8.0-SNAPSHOT/tachyon-parent-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/dependency-reduced-pom.xml to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/clients/client/target/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-client/0.8.0-SNAPSHOT/tachyon-client-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/pom.xml to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/shell/target/tachyon-shell-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-shell/0.8.0-SNAPSHOT/tachyon-shell-0.8.0-SNAPSHOT-javadoc.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/target/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jar to org.tachyonproject/tachyon-underfs-glusterfs/0.8.0-SNAPSHOT/tachyon-underfs-glusterfs-0.8.0-SNAPSHOT-javadoc.jarchannel stopped
Test FAILed.

swatDong commented:
Jenkins, test it please.

swatDong commented:
Jenkins, test this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5006/
Test PASSed.

swatDong commented:
@gpang I've addressed the comments and updated with master branch.
    1) Print the help in java by using `org.apache.commons.cli.HelpFormatter`
    2) Kill/restart Master only in java for multiple times

gpang commented on examples/src/main/java/tachyon/examples/JournalCrashTest.java:
```diff
@@ -0,0 +1,387 @@
+  private static boolean parseInputArgs(String[] args) {
+    Options options = new Options();
+    options.addOption("help", false, "Show help for this test");
+    options.addOption("crashes", true, "Repeat times to kill/restart Master");
```
I think the way crashes are done now will not stress journal in a variety of situations, since the crashes all happen right after another. I think another way to do this is to define the total time to run the test for, and the maximum time a master should ever be alive. This means I could run the test for 24 hours, and if I want the master to be up for at most 1 hour, the test will kill the master after a random time (less than an hour). This means in the 24 hour test, I know the master would have been killed at least 24 times, but with random intervals.

swatDong replied:
That sounds good. I'll update to the random crash.

gpang commented:
I just had a suggestion for how the crashes work. Thanks!

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/Tachyon-Pull-Request-Builder/5059/
Test PASSed.

swatDong commented:
Have made a better crash way. Now the master will be killed with random intervals, and check status for each crash.

gpang commented:
LGTM
